debezium-core/src/main/java/io/debezium/relational/topic/TopicMapper.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational.topic;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+import io.debezium.relational.Table;
+
+import java.util.Map;
+
+/**
+ * @author David Leibovic
+ */
+public abstract class TopicMapper {
+
+    /**
+     * Get the name of the topic given for the table.
+     *
+     * @param topicPrefix prefix for the topic
+     * @param table the table that we are getting the topic name for
+     * @return the topic name; may be null if this strategy could not be applied
+     */
+    abstract public String getTopicName(String topicPrefix, Table table);
```

Comment:
I have not come up with a solution to enforce that custom `TopicMapper` implementations do not map tables with two different schemas to the same topic.

The only thing I could come up with is something like this, which merely enforces that all topic names have some non-null component that is based off the actual table name:
```java
public abstract class TopicMapper {

    /**
     * Get the name of the topic given for the table.
     *
     * @param topicPrefix prefix for the topic
     * @param table the table that we are getting the topic name for
     * @return the topic name; may be null if this strategy could not be applied
     */
    final public String getTopicName(String topicPrefix, Table table) {
        String prefixComponent = getPrefixComponent(topicPrefix);
        String catalogComponent = getCatalogComponent(table.id().catalog());
        String schemaComponent = getSchemaComponent(table.id().schema());
        String tableComponent = getTableComponent(table.id().table());

        if (tableComponent == null || tableComponent == "") {
            // TODO: do some error -- the topic must have some table specific component!
        }

        String topicName = "";
        if (prefixComponent != null) {
            topicName += topicName + "." + prefixComponent;
        }
        if (catalogComponent != null) {
            topicName += topicName + "." + catalogComponent;
        }
        if (schemaComponent != null) {
            topicName += topicName + "." + schemaComponent;
        }
        topicName += topicName + "." + tableComponent;
        return topicName;
    }
    
    abstract protected String getPrefixComponent(String topicPrefix);
    abstract protected String getCatalogComponent(String catalog);
    abstract protected String getSchemaComponent(String schema);
    abstract protected String getTableComponent(String table);
}
```

but this feels clunky and still hard to enforce.